File: 0290-word-pattern/0290-word-pattern.ts
function wordPattern(pattern: string, s: string): boolean {
  let words = s.split(' ');
  let mapper1 = new Map();
  let mapper2 = new Map();
    
    if (words.length !== pattern.length) return false

  for (let i = 0; i < pattern.length; i++) {
    const char = pattern[i];
    const word = words[i];

    if (!mapper1.has(char) && !mapper2.has(word)) {
      mapper1.set(char, word);
      mapper2.set(word, char);
    } else if (mapper1.get(char) !== word || mapper2.get(word) !== char) {
      return false;
    }
  }

  return true;
}

File: 0290-word-pattern/0290-word-pattern.py
class Solution:
    def wordPattern(self, pattern: str, s: str) -> bool:
        words = s.split(" ")
    
        if len(words) != len(pattern): 
            return False

        word_map = {}
        char_map = {}


        for word,char in zip(words, pattern):
            if word not in word_map and char not in char_map:
                word_map[word] = char
                char_map[char] = word

            elif word != char_map.get(char) or char != word_map.get(word):
                return False

        return True